R/util.R

```diff
@@ -0,0 +1,279 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#' Joins two DQD results
+#' 
+#' @param jsonPath.1 the path to the first DQD json results file
+#' @param jsonPath.2 the path to the second DQD json results file
+#' 
+#' @return A tibble with the joined dqd results, one row per DQD check.
+.joinDqdResults <- function(jsonPath.1, jsonPath.2, suffixes = c(".1", ".2")){
+  # List all differences
+  # ... between OLD
+  result_old <- jsonlite::fromJSON(jsonPath.1)
+  check_results_old <- tibble(result_old$CheckResults)
+  
+  # ... and NEW
+  result_new <- jsonlite::fromJSON(jsonPath.2)
+  check_results_new <- tibble(result_new$CheckResults)
+  
+  # ... and join
+  combined_results <- check_results_old %>%
+    left_join(check_results_new,
+              by=c("CHECK_NAME", "CDM_TABLE_NAME", "CDM_FIELD_NAME",
+                   "CONCEPT_ID", "UNIT_CONCEPT_ID"),
+              suffix=suffixes)
+  
+  return(combined_results)
+}
+
+#' Create an interactive scatter plot comparing two DQD results
+#' 
+#' Matches checks from the two provided DQD results and saves two outputs:
+#' \itemize{
+#'  \item a table with all checks that are different
+#'  \item an interactive plot (html) with old fail percentage versus new fail
+#'  percentage
+#' } 
+#' This makes it easy to identify differences between subsequent runs of DQD.
+#' Nothing will be created if there are no differences between the given DQD results.
+#' See also: \link[TBD]{TBD - link to abstract}
+#' 
+#' @param jsonPath.old the path to the old DQD json results file
+#' @param jsonPath.new the path to the new DQD json results file
+#' @param savingDir (optional) the path to the folder where the output should be written.
+#'                  if not given, no output will be written.
+#' 
+#' @author Elena Garcia Lara
+#' @author Maxim Moinat
+#' 
+#' @return An interactive plotly figure or nothing if no differences are found.
+#' @export
+#' 
+#' @examples 
+#' \dontrun{
+#'   plotCompareDqdResults("dqd_results_1.json", "dqd_results_2.json", "output")
+#' }
+plotCompareDqdResults <- function(jsonPath.old, jsonPath.new, savingDir = NA){
+  combinedResult <- .joinDqdResults(jsonPath.old, jsonPath.new, suffixes = c(".old", ".new"))
+  
+  # Only keep changed
+  combinedResult <- combinedResult %>%
+                      filter(PCT_VIOLATED_ROWS.old != PCT_VIOLATED_ROWS.new)
+  
+  # When no difference found, exit function
+  if(nrow(combinedResult)==0){
+    stop("No differences found.")
+  }
+  
+  # Visualization
+  p <- combinedResult %>%
+    mutate(       
+      fail_status = ifelse(FAILED.old, 
+                           ifelse(FAILED.new, "Fail→Fail", "Fail→Pass"),
+                           ifelse(FAILED.new, "Pass→Fail", "Pass→Pass")
+                           ),
```

Comment:
```suggestion
                           ifelse(FAILED.new, "Fail-to-Fail", "Fail-to-Pass"),
                           ifelse(FAILED.new, "Pass-to-Fail", "Pass-to-Pass")
                           ),
```

Love the arrows! But they give an error when I make the package :(